core/execution/collateral.go

```diff
@@ -86,11 +86,14 @@ func (m *Market) transferRecheckMargins(ctx context.Context, risk []events.Risk)
 	}
 	mID := m.GetID()
 	evts := make([]events.Event, 0, len(risk))
+	var err error
+	var closed events.Margin
 	for _, r := range risk {
+		var tr *types.LedgerMovement
 		responses := make([]*types.LedgerMovement, 0, 1)
-		tr, closed, err := m.collateral.MarginUpdateOnOrder(ctx, mID, r)
+		tr, closed, err = m.collateral.MarginUpdateOnOrder(ctx, mID, r)
 		if err != nil {
-			return err
+			break
```

Comment:
@vegaprotocol/core probably worth continuing here instead and logging the error somehow. Any thoughts? Should I create something like:
```
errors.New("one or more margin updates failed during margin recheck" 
```
and hide away the details coming in the error returned by `.MarginUpdateOnOrder` or somehow combine the errors? (or leave as is?)

Reply:
Hmm, I need to go back through everything I was doing when adding this function, but I believe that the point at which this is called, it technically shouldn't be possible for errors occurring here (ie balances should be fine). That would mean that, if the collateral engine returns an error, that means that we're dealing with margin transfer requests for parties who don't exist in the collateral engine, and things are going quite dramatically wrong.

Reply:
OK, that may well be the case given there's a distressed party there. I can take a look and try to get to the bottom of it.

Reply:
@EVODelavega I'd be keen to get this merged as is (it's same functionality, just reports the event correctly now) and deal with the rest as part of this, what do you think: https://github.com/vegaprotocol/vega/issues/7543 ?